### DISCUSSION TITLE
[Discussion][Document] The content of libraries implementation status page

### DISCUSSION THREAD

**Discussion by acezen** [apache/incubator-graphar#378] [Ideas]
as issue #352 #362 describe, we need to provide an implementation status to indicated the format implementation status of GraphAr libraries.
I draft a page about the implementation status and maybe I'm missing something important.  So I want to discuss first.

**Comment by acezen** [comment]
# Implementation Status

The following tables summarize the features available in the various official GraphAr libraries. All libraries currently follow version 1.0.0 of the GraphAr format.

## Property

| Property feature | C++ | Java | Scala | Python |
|------------------|-----|------|-------|--------|
| primary key      | ✓   | ✓    | ✓     | ✓      |
| nullable         | ✓   |      | ✓     | ✓ (2)  |

**Notes:**
- \(1) Through py4j bindings to the GraphAr spark library.

## Property Group

| Property Group | C++ | Java | Scala | Python |
|----------------|-----|------|-------|--------|
| create         | ✓   | ✓    | ✓     | ✓      |
| add            | ✓   | ✓    | ✓     | ✓      |
| delete         |     |      |       |        |

## Adjacency List

| Adjacency List | C++ | Java | Scala | Python |
|----------------|-----|------|-------|--------|
| CSR            | ✓   | ✓    | ✓     | ✓      |
| CSC            | ✓   | ✓    | ✓     | ✓      |
| COO            | ✓   | ✓    | ✓     | ✓      |

## Data Types

### Primitive

| Data type | C++ | Java | Scala | Python |
|-----------|-----|------|-------|--------|
| Boolean   | ✓   | ✓    | ✓     | ✓      |
| Int32     | ✓   | ✓    | ✓     | ✓      |
| Int64     | ✓   | ✓    | ✓     | ✓      |
| Float     | ✓   | ✓    | ✓     | ✓      |
| Double    | ✓   | ✓    | ✓     | ✓      |
| String    | ✓   | ✓    | ✓     | ✓      |
| Date      |     |      |       |        |
| Time      |     |      |       |        |
| Timestamp |     |      |       |        |

### Nested

| Data type | C++ | Java | Scala | Python |
|-----------|-----|------|-------|--------|
| List      | ✓ (1) |      |       |        |

**Notes:**
- \(1) The list data type is not supported by CSV payloads.

## Payload Data File Formats

| Format  | C++ | Java | scala | python  |
|---------|-----|------|-------|---------|
| Avro    |     |      |       |         |
| CSV     | R/W | R (1) | R/W   | R/W (2) |
| ORC     | R/W | R (1) | R/W   | R/W (2) |
| Parquet | R/W | R (1) | R/W   | R/W (2) |

**Notes:**
- *R* = Read supported
- *W* = Write supported
- \(1) Through FastFFI bindings to the GraphAr C++ library.
- \(2) Through py4j bindings to the GraphAr spark library.

## Graph Metadata

| Graph Metadata | C++ | Java | Scala | Python |
|----------------|-----|------|-------|--------|
| Graph name     | ✓   | ✓    | ✓     | ✓      |
| Version        | ✓   | ✓    | ✓     | ✓      |
| Extra info     | ✓   |      |       |        |

**Comment by acezen** [comment]
cc\ @lixueclaire @Thespica @SemyonSinchenko Please let me know your opinion about the draft.

**Reply by Thespica** [reply]
LGTM

Btw, Should we add diff of schema in Java SDK as we had discussed here #346 ?

**Reply by acezen** [reply]
I think the JAVA SDK should update to align the schema, the diff is just a ad-hot solution for passing the CI

**Comment by lixueclaire** [comment]
> cc\ @lixueclaire @Thespica @SemyonSinchenko Please let me know your opinion about the draft.

Hi, @acezen , this proposal looks promising. I recommend adding brief details on the following: **Compression method** and **Supported Reader/Writer Levels** (low-level, mid-level, high-level).

**Reply by acezen** [reply]
this is a good advice!
